src/herder/TransactionQueue.cpp

```diff
@@ -187,6 +205,17 @@ TransactionQueue::getState(AccountID const& accountID) const
 void
 TransactionQueue::shift()
 {
+    mBannedTransactions.erase(--mBannedTransactions.end());
+    mBannedTransactions.emplace_front();
+
+    for (auto const& map : mPendingTransactions.back())
+    {
+        for (auto const& toBan : map.second->mTransactions)
+        {
+            mBannedTransactions[0].insert(toBan.first);
```

Comment:
move `mBannedTransactions[0]` outside the loops, probably just a reference to `mBannedTransactions.front()`